### PR TITLE
Widget preivew not working if widget registered via a instance

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Description
Details on this ticket can be found here - https://core.trac.wordpress.org/ticket/53427

This ticket introduces a new filter called `wp_get_legacy_widget_key` which means that we do not have hardcode references to `gutenberg_get_widget_object` into core. 

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/widgets/src/blocks/legacy-widget/index.php [hunk 1]
```diff
@@ -24,13 +24,14 @@ function render_block_core_legacy_widget( $attributes ) {
 		return '';
 	}
 
-	if ( method_exists( $wp_widget_factory, 'get_widget_object' ) ) {
-		$widget_object = $wp_widget_factory->get_widget_object( $attributes['idBase'] );
+	$base_id = $attributes['idBase'];
```
Comment (noisysocks) [inline packages/widgets/src/blocks/legacy-widget/index.php]: This should be `$id_base` to match what it's called in other parts of the codebase.

**Review by noisysocks — APPROVED** [review]
Tested that previewing a Legacy Widget works when running plugin against `trunk` and `5.7`. LGTM 👍

File: packages/widgets/src/blocks/legacy-widget/index.php (hunk 1 above)
Comment (spacedmonkey) [inline packages/widgets/src/blocks/legacy-widget/index.php]:
```suggestion
	$id_base = $attributes['idBase'];
```

File: packages/widgets/src/blocks/legacy-widget/index.php [hunk 2]
```diff
@@ -24,13 +24,14 @@ function render_block_core_legacy_widget( $attributes ) {
...
+	if ( method_exists( $wp_widget_factory, 'get_widget_key' ) ) {
+		$widget_key = $wp_widget_factory->get_widget_key( $base_id );
```
Comment (spacedmonkey) [inline packages/widgets/src/blocks/legacy-widget/index.php]:
```suggestion
		$widget_key = $wp_widget_factory->get_widget_key( $id_base );
```

File: packages/widgets/src/blocks/legacy-widget/index.php [hunk 3]
```diff
@@ -24,13 +24,14 @@ function render_block_core_legacy_widget( $attributes ) {
...
 	} else {
-		$widget_object = gutenberg_get_widget_object( $attributes['idBase'] );
+		$widget_key = gutenberg_get_widget_key( $base_id );
```
Comment (spacedmonkey) [inline packages/widgets/src/blocks/legacy-widget/index.php]:
```suggestion
		$widget_key = gutenberg_get_widget_key( $id_base );
```

File: packages/widgets/src/blocks/legacy-widget/index.php (hunk 1 above)
Comment (spacedmonkey) [reply packages/widgets/src/blocks/legacy-widget/index.php]: @noisysocks Fixed!

**Comment by adamziel** [comment]
The failure looks unrelated, I just restarted the job.

**Pull request merged**